message/message.go

```diff
@@ -0,0 +1,195 @@
+package message
+
+import (
+	"google.golang.org/protobuf/proto"
+)
+
+// IBFTMessage defines the 4 message types used in the IBFT 2.0 protocol
+// to reach network-wide consensus on some proposal for a particular sequence (height)
+type IBFTMessage interface {
+	*MsgProposal | *MsgPrepare | *MsgCommit | *MsgRoundChange
+}
+
+// Message is an opaque wrapper for the IBFT consensus messages. See IBFTMessage for concrete type definitions
+type Message interface {
+	GetInfo() *MsgInfo
+}
+
+// Signer is identified by its Address and used to generate signature for arbitrary payload
+type Signer interface {
+	// Address returns the public ID of Signer
+	Address() []byte
+
+	// Sign returns the signature generated from data
+	Sign(data []byte) []byte
+}
+
+// SignatureVerifier validates Signer signatures
+type SignatureVerifier interface {
+	// Verify checks if the signature of the message is valid
+	Verify(signer, digest, signature []byte) error
```

Comment:
Why is this a different interface from `Signer`?

The idea is `SignatureVerifier.Verify(..., Signer.Sign(...)) == true`